filters/builtin/inlinecontentifstatus.go

```diff
@@ -0,0 +1,92 @@
+package builtin
+
+import (
+	"bytes"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/zalando/skipper/filters"
+)
+
+type inlineContentIfStatus struct {
+	statusCode int
+	text       string
+	mime       string
+}
+
+// Creates a filter spec for the inlineContent() filter.
+//
+//     * -> inlineContentIfStatus(401, "{\"foo\": 42}", "application/json") -> "https://www.example.org"
+//
+// It accepts three arguments: the statusCode code to match, the content and the optional content type.
+// When the content type is not set, it tries to detect it using http.DetectContentType.
+//
+// The filter replaces the response coming from the backend or the following filters.
+//
+func NewInlineContentIfStatus() filters.Spec {
+	return &inlineContentIfStatus{}
+}
+
+func (c *inlineContentIfStatus) Name() string { return InlineContentIfStatusName }
+
+func (c *inlineContentIfStatus) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) < 2 || len(args) > 3 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	var (
+		f  inlineContentIfStatus
+		ok bool
+	)
+
+	f.statusCode, ok = args[0].(int)
+	if !ok {
+		floatStatusCode, ok := args[0].(float64)
+		if !ok {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+		f.statusCode = int(floatStatusCode)
+	}
+
+	if f.statusCode < 100 || f.statusCode >= 600 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	f.text, ok = args[1].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if len(args) == 3 {
+		f.mime, ok = args[2].(string)
+		if !ok {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	} else {
+		f.mime = http.DetectContentType([]byte(f.text))
+	}
+
+	return &f, nil
+}
+
+func (c *inlineContentIfStatus) Request(filters.FilterContext) {}
+
+func (c *inlineContentIfStatus) Response(ctx filters.FilterContext) {
+	if ctx.Response().StatusCode != c.statusCode {
+		return
+	}
+	rsp := ctx.Response()
+
+	err := rsp.Body.Close()
```

Comment:
I am not sure if I need to do this or something else with the original request...

Reply:
We definitely need to do this. we learned the hard way. :smile:

Reply:
@ruiaraujo Is this sufficient though or do I need to explicitly consume the body to avoid weird result for the backend server?

Reply:
we need to close the response body as for the contract with the library used on our side (net/http, see the Body field docs in https://godoc.org/net/http#Response)

about cosuming the response body: that's not mandatory, a client can just refuse to consume it. Sometimes, with long streams, it may result in logspam on the server side, but it's still better not to put that load on the pipe, IMHO. In addition to this, I expect that in the typical use case of this filter, the body won't be too long, which means that probably all the bytes are already buffered on the client side, so we can just safely discard it. So I suggest to not consume it.